sbol_utilities/helper_functions.py

```diff
@@ -105,6 +105,46 @@ def url_to_identity(url: str) -> str:
     return f'{split[0]}/{sbol3.string_to_display_id(split[1])}'
 
 
+def is_dna_part(obj: sbol3.Document) -> bool:
+    """Check if an SBOL Component is a DNA Part, i.e., having type 'SBO:DNA', and exactly 1 sequence property
+
+    :param obj: Sbol3 Document to be checked
+    :return: true if dna part
+    """
+    # must have only 1 type - of dna
```

Comment:
Note that these can likely be done more compactly and elegantly via an "any" applied to a generator

Reply:
Alright, I missed it in the sep, 
> Note that these can likely be done more compactly and elegantly via an "any" applied to a generator
also, I didn't quite get what you meant here.

Reply:
You can use a form like: `any(type.is_a(tyto.SBO.DNA) for type in component.types)`

Reply:
Actually, I'm not sure how it worked for you before, given that it was using `x.type` rather than `x.types`. I think you need to turn the type into a tyto term in order to do your comparison.

Reply:
Actually, the data-model specification had the field as "type" which I found was implemented instead as "types" in pysbol, so before adding the tests, I had made these conversions (`type -> types`, `role -> roles`, `hasSequence -> sequences`), so it passed without issues. I will try turning it into a tyto term though.

Reply:
That's correct: for multi-valued properties, RDF serializes each instance of a value separately, and thus the name is singular. When we collect them into a list in python, however, that construct may have multiple and thus receives a plural name. Likewise, there isn't the same namespace reason to prepend with "has".

Reply:
Oh right. 
Actually tyto's `get_term_from_uri` is not returning a term object to apply `is_a` on, I am instead getting a string, so I have placed a different type of comparison test.
I think my local test run didn't throw the same error as the actions had thrown because maybe the sbol_utilities package which the tests were using were not updated unless I reinstall using setup.py right?

Reply:
What you've got now also isn't the right form, since it only tests for string equality. What you want is something of the form:
```
any(tyto.SBO.deoxyribonucleic_acid.is_ancestor_of(t) for t in types)
```